indexer/bag_indexer/src/test/scala/uk/ac/wellcome/platform/archive/indexer/bag/BagIndexerWorkerTest.scala

```diff
@@ -2,41 +2,206 @@ package uk.ac.wellcome.platform.archive.indexer.bag
-import uk.ac.wellcome.platform.archive.indexer.bags.{
-  BagIndexer,
-  BagsIndexConfig
-}
 import uk.ac.wellcome.platform.archive.indexer.bags.models.IndexedStorageManifest
-import uk.ac.wellcome.platform.archive.indexer.elasticsearch.Indexer
+import uk.ac.wellcome.platform.archive.indexer.bags.{BagIndexer, BagIndexerWorker, BagsIndexConfig}
+import uk.ac.wellcome.platform.archive.indexer.elasticsearch.{Indexer, IndexerWorker}
+import uk.ac.wellcome.storage.store.HybridStoreEntry
+import uk.ac.wellcome.storage.store.memory.MemoryVersionedStore
+import uk.ac.wellcome.storage.{DoesNotExistError, ReadError, StoreReadError}
 
 import scala.concurrent.ExecutionContext.Implicits.global
 
+
 class BagIndexerWorkerTest
-    extends IndexerWorkerTestCases[StorageManifest, IndexedStorageManifest]
-    with StorageManifestGenerators {
+  extends IndexerWorkerTestCases[
+    KnownReplicasPayload,
+    StorageManifest,
+    IndexedStorageManifest
+  ]
+    with StorageManifestGenerators
+    with PayloadGenerators
+    with IngestGenerators
+    with StorageManifestVHSFixture {
 
   override val mapping: MappingDefinition = BagsIndexConfig.mapping
 
-  override def createT: (StorageManifest, String) = {
-    val storageManifest = createStorageManifest
+  val version: BagVersion = BagVersion(1)
+  val ingest: Ingest = createIngestWith(version = Some(version))
+  val pipelineContext: PipelineContext = PipelineContext(ingest)
+
+  val bagInfo = createBagInfoWith(externalIdentifier = ingest.externalIdentifier)
+
+  val storageManifest: StorageManifest = createStorageManifestWith(
+    ingestId = ingest.id,
+    space = ingest.space,
+    version = version,
+    bagInfo = bagInfo
+  )
+  val payload: KnownReplicasPayload = createKnownReplicasPayloadWith(
+    context = pipelineContext,
+    version = version
+  )
 
-    (storageManifest, storageManifest.id.toString)
+  override def createT: (KnownReplicasPayload, String) = {
+    val bagId = BagId(
+      space = payload.storageSpace,
+      externalIdentifier = payload.externalIdentifier
+    )
+
+    (payload, bagId.toString)
   }
 
-  override def createIndexer(
-    index: Index
-  ): Indexer[StorageManifest, IndexedStorageManifest] =
+  def createIndexer(
+                     index: Index
+                   ): Indexer[StorageManifest, IndexedStorageManifest] =
     new BagIndexer(
       client = elasticClient,
       index = index
     )
 
   override def convertToIndexed(
-    manifest: StorageManifest
-  ): IndexedStorageManifest =
-    IndexedStorageManifest(manifest)
+                                 payload: KnownReplicasPayload
+                               ): IndexedStorageManifest = {
+    IndexedStorageManifest(storageManifest)
+  }
+
+  override def withIndexerWorker[R](index: Index, queue: SQS.Queue)(
+    testWith: TestWith[
+      IndexerWorker[KnownReplicasPayload, StorageManifest, IndexedStorageManifest],
+      R
+    ]
+  )(implicit decoder: Decoder[KnownReplicasPayload]): R = {
```

Comment:
Ditto above. Could this be pushed into a shared fixture? Looks identical.

Reply:
You'd think. It's possible, but the inheritance here makes it awkward to compose sensibly. Probably needs a re-jig, will rethink in a future PR.